test/libsolidity/SemanticTest.cpp

```diff
@@ -100,6 +109,12 @@ SemanticTest::SemanticTest(string const& _filename, langutil::EVMVersion _evmVer
 
 	parseExpectations(m_reader.stream());
 	soltestAssert(!m_tests.empty(), "No tests specified in " + _filename);
+
+	if (_evmVersion == EVMVersion{})
+	{
+		m_compiler.setVersionType(CompilerStack::VersionType::Empty);
+		m_compiler.setMetadataHash(CompilerStack::MetadataHash::None);
+	}
```

Comment:
I see that it was suggested by @ekpyron in https://github.com/ethereum/solidity/pull/10863#discussion_r570880001.

So yeah, the change was good but I think that making it depend on `_enforceGasCost` would produce the same effect and be semantically more correct (also, less chance that it will break if we suddenly decide to check gas expectations on more than one version).